app/controllers/topics_controller.rb

```diff
@@ -17,11 +20,31 @@ def create
   end
 
   def show
-    @topic = Topic.find(params[:id])
+  end
+
+  def destroy
+    Topic.find(params[:id]).destroy
+    redirect_to topics_path
+  end
+
+  def edit
+  end
+
+  def update
+    if @topic.update(topic_params)
+      redirect_to topic_path
+    else
+      @topic.name = Topic.find(params[:id]).name
```

Comment:
You don't need this line. You want to present the topic in its edited form, so the user can see the problem.

Reply:
(re your reply posted as a PR comment)

The main problem from a technical perspective is that you are doing two `find`s. That's two database queries, where only one is required.

You *could* use `ActiveModel::Dirty` as Iain suggested.
```
@topic.restore_attributes([:name])
```

...but why bother and what problem are you really solving? 

The reason the user can't identify the topic without a name is that there is no other data in the page, which means that there are no other edits to worry about and they can just navigate away from the page (e.g. go back to the topics#index page) and all is good (i.e. the name is never changed).

And so what if they enter a completely different topic name at this stage; the only thing identifying the topic is the name anyway. You don't currently have any linked tags or videos that need to match the topic name, and when you do, you'll have to worry about the issue of changing the topic name full stop (i.e. not just blanking it). Remember [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

Basically, this is probably its own feature. Whenever you add something as a nice little extra onto a feature, you introduce a whole load of complexity that isn't required, and often your quick fix doesn't fully/properly address the thing you're trying to address. T

I'd say think about what you want to happen here and why and capture it as a new story, and for this pull request, remove the line.